### ISSUE TITLE
Fix issues with #227

### ISSUE THREAD

**Issue by mxr576** [apigee/apigee-edge-drupal#234]
- [ ] Missing update hook
- [ ] Displaying an exception message as a message to an end-user never going to be translatable.
- [ ] Only admins has access to run the developer sync, we should not suggest simple authenticated users to run the developer sync as a fix.

Reverting #227 and re-evaluating possible options is also a possible fix for these.

Displaying translatable error messages from exceptions is not easy, but we already have something in the module: [KeyProviderRequirementsException](https://github.com/apigee/apigee-edge-drupal/blob/8.x-1.0/src/Exception/KeyProviderRequirementsException.php). 

Original post-merge review comments are here: https://github.com/apigee/apigee-edge-drupal/pull/227

Also related to #233 .

**Comment by cnovak** [comment]
For "missing update hook", the feature will not turned on by default for persons that already have enabled the module. However, they can turn this feature on if needed.  I do not see a strong need for this.

For the exception not being translatable, this is meant to help a user when first configuring the module, but could be an enhancement.  Opened #235 to address this.

For the "Only admins has access to run the developer sync", we have had a lot of issues with first time configuration of this module due to the Edge developer not being automatically created if the user does not exist in the Edge API.  This message is a quick fix to give the person installing the module an idea on what is going wrong.

**Comment by mxr576** [comment]
> For "missing update hook", the feature will not turned on by default for persons that already have enabled the module. However, they can turn this feature on if needed. I do not see a strong need for this.

@cnovak
TL;DR; Then set it to FALSE in an update hook.

Okay, so let me explain in details.
Since the module is stable it must keep the existing installations up to date. This also includes the active configuration on existing sites. You introduced a new config in #227  and its type according to [the config schema is boolean](https://github.com/arshad/apigee-edge-drupal/blob/36444eabbfc39b659e7b65efc1937205ee452617/config/schema/apigee_edge.schema.yml#L17). :heavy_check_mark: In every new installation[ its value will be `true` ](https://github.com/arshad/apigee-edge-drupal/blob/36444eabbfc39b659e7b65efc1937205ee452617/config/install/apigee_edge.error_page.yml#L6)which is a boolean value. :heavy_check_mark: Although, without an update hook, in existing sites this config's value will be `NULL`. :heavy_minus_sign: So the current `if ($this->configFactory->get('apigee_edge.error_page')->get('error_page_debug_messages'))` may works at this moment, because `NULL` automatically casted to `FALSE` in this case, but `NULL !== FALSE` therefore if someone writes `$this->configFactory->get('apigee_edge.error_page')->get('error_page_debug_messages') === FALSE` (why not? it should be a boolean...) its code won't work on an existing installation without the suggested update hook.

>  this is meant to help a user when first configuring the module
> we have had a lot of issues with first time configuration of this module due to the Edge developer not being automatically created

Yes, I am very well aware of this problem, this is an old one and I described plenty of possible options and quick workarounds for this in #115. Because this problem occurs mostly with UID1, if you wanted to implement a quick fix, you could have added just warning to README-s and display a warning message on the Authentication configuration form that developer sync must be executed whenApigee Edge credentials  *changes* - yes, it could cause problems later as well, not just after the initial configuration. (Probably this warning message would have been inherited by the Kickstarter as well if it exposes the Authentication form as-is, if not you could have added the warning message there as well.)

The merged solution may seem to provide a quick fix for one particular problem, but also introduces more that I detailed above. Thanks for the implementation, not just UID1-s gets warned about the user sync, but any other users who do not have a developer account on Apigee Edge _any time_.

**Comment by cnovak** [comment]
@mxr576 I opened #237 for the hook info you gave, thanks.
